### ISSUE TITLE
'length' is declared here

### ISSUE THREAD

**Issue by Xenfo** [modulz/stitches#947]
# Bug report

## Describe the bug

`createStitches` `styled` method requires `length` field to be set. `length` is part of `String` object.

## To Reproduce

```ts
export const { styled } = createStitches({
  prefix: 'fluxui'
});
```
```tsx
export const StyledIndicator = styled('div', {
  color: '$gray50',
  width: 15,
  height: 15
});
```
```
Argument of type '{ color: "$gray50"; width: number; height: number; }' is not assignable to parameter of type 'RemoveIndex<CSS<{}, { borderWeights: { light: string; normal: string; bold: string; extrabold: string; black: string; }; borderWidths: { default: string; 0: string; 2: string; 4: string; 8: string; }; breakpoints: { ...; }; ... 9 more ...; zIndices: { ...; }; }, DefaultThemeMap, {}>> & ... 6 more ... & { ...; }'.
  Property 'length' is missing in type '{ color: "$gray50"; width: number; height: number; }' but required in type '{ toString: () => string; readonly length: string | number | {} | CSS<{}, { borderWeights: { light: string; normal: string; bold: string; extrabold: string; black: string; }; borderWidths: { default: string; 0: string; 2: string; 4: string; 8: string; }; ... 10 more ...; zIndices: { ...; }; }, DefaultThemeMap, {}> |...'.ts(2345)
lib.es5.d.ts(503, 14): 'length' is declared here.
```

## Expected behavior

Not to have to enter `length`.

## System information

- OS: Linux
- Version of Stitches: 1.2.6
- Version of Node.js: 17.4.0

**Comment by hadihallak** [comment]
Would you mind creating a github repo or a codesandbox with the issue and linking it here?

**Comment by Xenfo** [comment]
Cloning this pr should do it, https://github.com/flux-ui/flux-ui/pull/15

**Comment by hadihallak** [comment]
@Xenfo Would really appreciate if you could provide something a bit more minimal as i'm not able to reproduce the issue in a [codesandbox](https://codesandbox.io/s/stoic-moore-nc21e?file=/src/App.tsx) and having a minimal reproduction of the issue will help us investigate this faster than going through a whole codebase

**Comment by Xenfo** [comment]
I can't seem to reproduce it in Codesandbox, maybe because my repo is a monorepo. The codebase is tiny though, 90% of files rn are config files. I'll try and make a smaller version of it on Codesandbox.

**Comment by Xenfo** [comment]
Here's a slimmed down branch, CodeSandbox was giving me problems. https://github.com/flux-ui/flux-ui/tree/example

**Comment by Xenfo** [comment]
Just realized VSCode was using a beta version of TypeScript, not Stitches fault, sorry!

**Comment by monsonjeremy** [comment]
I think we should re-open this. I noticed this issue with `typescript@^4.6.0`

**Comment by Sec-ant** [comment]
I encountered this issue too when using creat-react-app with typescript and stitches. It was all well but suddenly threw typecheck errors regarding `length` not defined. I wasn't sure what caused this problem.

**Comment by fmal** [comment]
@hadihallak i can also confirm that this issue is present with `typescript@4.6.2`, please consider reopening

**Comment by csantos-nydig** [comment]
Seeing the same error when I I tried to upgrade to `typescript@4.6.2`

**Comment by hmbrg** [comment]
On what `typescript` version is this bug not present? I've tried all the last version, nothing worked.

**Comment by norman-ags** [comment]
Experiencing the same issue with a monorepo setup

**Comment by lcoalves** [comment]
Experiencing the same issue with a monorepo setup.

**Comment by PookMook** [comment]
Same here, typescript 4.5.4 compile ok but typescript bundled in vscode (4.6.2) is not happy about it

**Comment by damianricobelli** [comment]
Experiencing the same issue with Typescript 4.5.4

**Comment by lcoalves** [comment]
I believe I was able to solve the problem. The VS Code was forcing Typescript to use version 4.6.2, I just changed it to use version 4.5.5 from my project.

@hadihallak The fact is that the problem actually occurs in version 4.6.2 with Stitches, but it worked correctly after this change.

Check out: https://www.loom.com/share/740bc375d2ea4767aedaaf7d06fd8e28

**Comment by lcoalves** [comment]
I believe I was able to solve the problem. The VS Code was forcing Typescript to use version 4.6.2, I just changed it to use version 4.5.5 from my project.

@hadihallak The fact is that the problem actually occurs in version 4.6.2 with Stitches, but it worked correctly after this change.

https://www.loom.com/share/740bc375d2ea4767aedaaf7d06fd8e28

**Comment by damianricobelli** [comment]
@lcoalves I confirm, this has solved it

**Comment by hadihallak** [comment]
Thanks for reopening the issue. Will be looking into it

**Comment by emdowu** [comment]
Problem occurs also with 4.4.4 - not present at node <= 14

**Comment by jgoz** [comment]
I might have a fix in https://github.com/modulz/stitches/pull/982. Tested it in our large monorepo and it all seems to be working with TS 4.6.

**Comment by emdowu** [comment]
> I might have a fix in #982. Tested it in our large monorepo and it all seems to be working with TS 4.6.

I have discovered it happens only with lerna.

**Comment by bartels** [comment]
> I might have a fix in https://github.com/modulz/stitches/pull/982. Tested it in our large monorepo and it all seems to be working with TS 4.6.

Can confirm #982 fixed the errors for me. I had to delete  `tsconfig.tsbuildinfo` files before re-running `tsc`

> I have discovered it happens only with lerna.

Not using lerna here, but a monorepo setup with yarn workspaces. 

I see no type errors in my `libs/ui` workspace that has the call to `createStitches` in it. Styled components in that workspace can use the resulting `styled` function without type errors.  However, other workspaces that import that `styled` function get this type error. #982 seems to fix those.

**Comment by bbg** [comment]
Hello friends,

I am developing a UI library with Stitches. I keep it as a Monorepo. However, it gives the error that I have attached the screenshot below while building. I tried what is mentioned here but no solution.

I don't understand what I'm missing or doing wrong :/

I would greatly appreciate it if anyone who has had this problem and solved it can help.

Codeb
<img width="574" alt="Screen Shot 2022-04-08 at 18 58 38" src="https://user-images.githubusercontent.com/13348876/162497619-76554529-20aa-47e8-a8cf-fbc7f7d01c07.png">
ase: https://codesandbox.io/s/recursing-jones-f95ywp?file=/package.json:515-530

**Comment by mntnoe** [comment]
We have also been blocked for a while in upgrading to TS 4.6 due to this issue. Maybe the TS issue only occurs for projects using `composite`/`--build`?

**Comment by SamSadotMerkle** [comment]
> I believe I was able to solve the problem. The VS Code was forcing Typescript to use version 4.6.2, I just changed it to use version 4.5.5 from my project.
> 
> @hadihallak The fact is that the problem actually occurs in version 4.6.2 with Stitches, but it worked correctly after this change.
> 
> Check out: https://www.loom.com/share/740bc375d2ea4767aedaaf7d06fd8e28

WOW! Amazing!. I have changed the TS version in VS code and the error is gone! Thanks for the video link.

**Comment by hevar** [comment]
It seems like this issue remains for Webstorm. Changing TS version to 4.5.5 resolves it and using 4.6.x will trigger it.

**Comment by norman-ags** [comment]
> It seems like this issue remains for Webstorm. Changing TS version to 4.5.5 resolves it and using 4.6.x will trigger it. 

Have you tried updating stitches?

**Comment by jgoz** [comment]
> Any workarounds with TypeScript 4.6.x?

This was fixed in the latest Stitches release
